### ISSUE TITLE
TUYA Human Presence Detector not supported - ZY_HPS01_5.8G

### ISSUE THREAD

**Issue by andersontd** [azerty9971/xtend_tuya#105]
I purchased this from Aliexpress without checking if it would be compatible with Home Assistant, HA see the device via the TUYA integration but shows as not supported. https://www.aliexpress.com/item/1005007334728544.html

Please let me know if you require any further details.

**Comment by azerty9971** [comment]
Hello,
On the device page can you download the diagnostics and post them here?
I'll add the support for it ;).

Have a nice day,
Azerty

**Comment by andersontd** [comment]
Hi Azerty

See attached.
[tuya-16746ad3fc60e87faf7f5d15e946df91-HPS 1-262f3675d5d3ea45b4ed1b8cd2dff19d.json](https://github.com/user-attachments/files/17096479/tuya-16746ad3fc60e87faf7f5d15e946df91-HPS.1-262f3675d5d3ea45b4ed1b8cd2dff19d.json)

**Comment by azerty9971** [comment]
I've read the diagnostic and the device doesn't push any statuses.
It might have more information if you add Tuya Cloud credentials:
https://github.com/azerty9971/xtend_tuya#adding-your-tuya-cloud-credentials

Could you add those credentials and re-export the diagnostic file from the Xtend Tuya integration? (where you downloaded the diagnostics, you'll have a 3 dots button that you can push to select which diagnostics to download :))

Have a nice day,
Azerty

**Comment by andersontd** [comment]
Hi
[TT-tuya-16746ad3fc60e87faf7f5d15e946df91-HPS 1-262f3675d5d3ea45b4ed1b8cd2dff19d.json](https://github.com/user-attachments/files/17117535/TT-tuya-16746ad3fc60e87faf7f5d15e946df91-HPS.1-262f3675d5d3ea45b4ed1b8cd2dff19d.json)

I have followed the youtube video and the json is different, hopefully it is what you need.

**Comment by azerty9971** [comment]
Unfortunately there is still no status being sent in the diagnostic file :/

Could you share the device with me in the Tuya app so that I can have a look from my side ?
Account: devun999999@gmail.com

Have a nice evening,
Azerty

**Comment by andersontd** [comment]
Hi

Just checking that you got the share link.

**Comment by azerty9971** [comment]
Hello,
I've checked the accounts on both Tuya and SmartLife and I haven't received your device :/.

Which app did you use?
Could you please retry the sharing?

Thanks in advance,
Azerty

**Comment by azerty9971** [comment]
Hello,
The latest beta implements this device:
![image](https://github.com/user-attachments/assets/a8cb7c72-371d-43f7-9503-34d1cba774ed)

To have the sensors, you need to have the Tuya IOT Cloud credentials in the integration.
If you don't see them, it means that something is wrong there ;). (from your diagnostic file, I guess there is something missing).

Please update to the latest BETA for this to work :).

Have a nice day,
Azerty

**Comment by andersontd** [comment]
Many thanks for your work on this.

Now for the dumb question, how do I install the beta?

Also, can you point me at any documentation so I can try and understand a bit more as I have a few Tuya devices and I think there is more they could do. i.e. Smart thermostat, I can switch modes but not set the temperature and I know it must be possible because it is available in the standard tile.

Thanks again

**Comment by andersontd** [comment]
I have worked out how to install the beta and I can now see the sensors. Awesome work. thanks again

Anything you can point me at to be a bit more self sufficient would be greatly appreciated.

**Comment by azerty9971** [comment]
Hello,
Sorry, I just saw your messages now. Glad you could figure it out :).

Regarding your other Tuya devices, to know if something (sensor, slider, button, ...) is implementable in HA, you just need to look at the diagnostic file:
- On the device page, click the "Download diagnostics" and open it with a JSON file viewer (Firefox for instance)
- Go and read the codes in the "function", "status_range" and "status" sections:
![image](https://github.com/user-attachments/assets/fe545103-c76e-4be4-8e17-f1935c58094f)

Typically, things are like this:
- Codes in function are things you can operate with (switch, setting, anything you can change from HA to the device)
- Codes in status_range are sensor data (things you can only read but not change)
- Codes in status are the current values of the codes in both function and status_range

Basically, if you have something in status that is not shown up on the device page in HA then it means your device has entities that could be supported but are not yet implemented (that's where I can help :) )
If you want to do the implementation yourself, then you can have a look at this example of adding support for HPS1:
https://github.com/azerty9971/xtend_tuya/commit/6b57bbcb836fe0236e76ed00ff7bc1ed4e85d85d

Hope this helps!!

Have a nice day,
Azerty

**Comment by andersontd** [comment]
Many thanks, I will take a look
